tests/JustSaying.UnitTests/Messaging/Compression/GzipMessageBodyCompressionTests.cs

```diff
@@ -0,0 +1,74 @@
+using JustSaying.Messaging.Compression;
+
+namespace JustSaying.UnitTests.Messaging.Compression
+{
+    public class GzipMessageBodyCompressionTests
+    {
+        private readonly GzipMessageBodyCompression _compression = new();
+
+        [Fact]
+        public void ContentEncoding_ShouldReturnGzipBase64()
+        {
+            Assert.Equal(ContentEncodings.GzipBase64, _compression.ContentEncoding);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("Hello, World!")]
+        [InlineData("This is a longer string with some special characters: !@#$%^&*()_+")]
+        public void Compress_ThenDecompress_ShouldReturnOriginalString(string original)
+        {
+            // Arrange
+
+            // Act
+            string compressed = _compression.Compress(original);
+            string decompressed = _compression.Decompress(compressed);
+
+            // Assert
+            Assert.Equal(original, decompressed);
+        }
+
+        [Fact]
+        public void Compress_ShouldReturnBase64EncodedString()
+        {
+            // Arrange
+            string input = "Test string";
+
+            // Act
+            string compressed = _compression.Compress(input);
+
+            // Assert
+            Assert.True(IsBase64String(compressed));
+        }
+
+        [Fact]
+        public void Decompress_WithInvalidBase64_ShouldThrowFormatException()
+        {
+            // Arrange
+            string invalidBase64 = "This is not a valid Base64 string";
+
+            // Act & Assert
+            Assert.Throws<FormatException>(() => _compression.Decompress(invalidBase64));
+        }
+
+        [Fact]
+        public void Compress_WithLargeString_ShouldCompressSuccessfully()
+        {
+            // Arrange
+            string largeString = new string('A', 1000000);  // 1 million 'A' characters
```

Comment:
Could also call the variable `oneMillionLetters` or something

```suggestion
            string largeString = new string('A', 1_000_000);
```